### PR TITLE
module 10 homework

### PR DESCRIPTION
**Pull request by pwrice**

### CODE REVIEW

File: module10/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+g) True or False: If you wrap code within a Task, it will always run on the main thread.
+
+False
+
+h) In SwiftUI, to resolve the issue of calling an asynchronous method in a nonconcurrent context, one can replace onAppear with ____________.
+
+
+the .task() view modifier
+
+i) True or Flase: In SwiftUI, view modifiers like onAppear inherently run code asynchronously.
```
Comment (waltertyree) [inline module10/README.md:36]: No, they run on the main thread unless you tell them to go to the background using `Task` or something similar.

File: module10/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+True
+
+
+j) The ____________ keyword indicates a function contains a suspension point.
```
Comment (waltertyree) [inline module10/README.md:41]: It's confusingly worded, but it is asking how can you know a function contains a suspensiont. Or...what do you put n the function declaration to let people know you're going to be messing around on multiple threads? so async.

File: module10/README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+async
+
+m) The AsyncSequence protocol requires defining the element type and providing an ______________.  
+
+AsyncIterator
+
+n) Task cancellation in Swift is ______________ in nature.
+
+Cooperative ? Hierarchical ?
```
Comment (waltertyree) [inline module10/README.md:60]: Cooperative

**Review by waltertyree — APPROVED** [review]
Unless I'm missing something, you never catch things that are thrown. You should get in the habit of catching somewhere in your code, closer to the throw the better since you'll be able to track  down issues faster. Future you will thank you. If you don't catch things that are thrown, then throws will crash your code and the stack trace will be in the main app file. This won't be very helpful. Look at everywhere you have a `do` or a `try`. These are places you can `catch` to keep your code from crashing the app.

File: module10/README.md (hunk 1 above)
Comment (pwrice) [reply module10/README.md:36]: Ah i misread this one. Yes my understanding was that all SwiftUI runs on the main thread by default.

**Review by waltertyree — APPROVED** [review]
Another thing you can do to polish, on the next codes, is make your own Error enum and practice throwing those, also looking at Logger() instead of using `print`. But just something to look at as you do more coding. No need to add them to this.

**Pull request merged**